src/Microsoft.Health.Fhir.Shared.Api/Features/Resources/Bundle/BundleHandler.cs

```diff
@@ -66,23 +76,67 @@ public async Task<BundleResponse> Handle(BundleRequest bundleRequest, Cancellati
         {
             var bundleResource = bundleRequest.Bundle.ToPoco<Hl7.Fhir.Model.Bundle>();
 
-            if (bundleResource.Type != Hl7.Fhir.Model.Bundle.BundleType.Batch)
+            await FillRequestLists(bundleResource.Entry);
+
+            if (bundleResource.Type == Hl7.Fhir.Model.Bundle.BundleType.Batch)
             {
-                throw new MethodNotAllowedException(Microsoft.Health.Fhir.Api.Resources.OnlyCertainBundleTypesSupported);
+                var responseBundle = new Hl7.Fhir.Model.Bundle
+                {
+                    Type = Hl7.Fhir.Model.Bundle.BundleType.BatchResponse,
+                };
+
+                await ExecuteAllRequests(responseBundle);
+                return new BundleResponse(responseBundle.ToResourceElement());
             }
+            else if (bundleResource.Type == Hl7.Fhir.Model.Bundle.BundleType.Transaction)
+            {
+                var responseBundle = new Hl7.Fhir.Model.Bundle
+                {
+                    Type = Hl7.Fhir.Model.Bundle.BundleType.TransactionResponse,
+                };
 
-            await FillRequestLists(bundleResource.Entry);
+                return await ExecuteTransactionForAllRequests(responseBundle);
+            }
 
-            var responseBundle = new Hl7.Fhir.Model.Bundle
+            throw new MethodNotAllowedException(Microsoft.Health.Fhir.Api.Resources.OnlyCertainBundleTypesSupported);
```

Comment:
We should update this message to say something like `Bundles of type '{type}' are not supported.`